src/main/java/org/jboss/pnc/deliverablesanalyzer/model/FinderResult.java

```diff
@@ -0,0 +1,279 @@
+import org.jboss.pnc.build.finder.koji.KojiLocalArchive;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.red.build.koji.model.xmlrpc.KojiArchiveInfo;
+
+public class FinderResult {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FinderResult.class);
+
+    @NotNull
+    @Valid
+    private Set<Build> builds;
+
+    @NotNull
+    @Valid
+    private Set<Artifact> notFoundArtifacts;
+
+    @NotNull
+    @Valid
+    private BuildStatistics statistics;
+
+    public FinderResult() {
+
+    }
+
+    public FinderResult(Map<BuildSystemInteger, KojiBuild> builds, String username) {
+        this.builds = getFoundBuilds(builds, username);
+        this.notFoundArtifacts = getNotFoundArtifacts(builds, username);
+        this.statistics = new BuildStatistics(getBuildsAsList(builds));
+    }
+
+    public Set<Build> getBuilds() {
+        return Collections.unmodifiableSet(builds);
+    }
+
+    public Set<Artifact> getNotFoundArtifacts() {
+        return Collections.unmodifiableSet(notFoundArtifacts);
+    }
+
+    public BuildStatistics getStatistics() {
+        return statistics;
+    }
+
+    private Set<Artifact> getNotFoundArtifacts(Map<BuildSystemInteger, KojiBuild> builds, String username) {
+        KojiBuild buildZero = builds.get(new BuildSystemInteger(0));
+        List<KojiLocalArchive> localArchives = buildZero.getArchives();
+        int numArchives = localArchives.size();
+        Set<Artifact> artifacts = new LinkedHashSet<>(numArchives);
+        int archiveCount = 0;
+
+        for (KojiLocalArchive localArchive : localArchives) {
+            archiveCount++;
+
+            Collection<String> files = localArchive.getFilenames();
+            Collection<Checksum> checksums = localArchive.getChecksums();
+            Artifact artifact = new Artifact();
+
+            artifact.setBuiltFromSource(Boolean.FALSE);
+
+            for (Checksum checksum : checksums) {
+                switch (checksum.getType()) {
+                    case md5:
+                        artifact.setMd5(checksum.getValue());
+                        break;
+                    case sha1:
+                        artifact.setSha1(checksum.getValue());
+                        break;
+                    case sha256:
+                        artifact.setSha256(checksum.getValue());
+                        break;
+                    default:
+                        break;
+                }
+            }
+
+            artifact.setFilesNotBuiltFromSource(new TreeSet<>(files));
+            artifact.setUsername(username);
+            artifact.setCreated(new Date());
+
+            artifacts.add(artifact);
+
+            LOGGER.info("Artifact: {} / {} ({})", archiveCount, numArchives, artifact.getFilesNotBuiltFromSource());
+        }
+
+        return Collections.unmodifiableSet(artifacts);
+    }
+
+    private Set<Build> getFoundBuilds(Map<BuildSystemInteger, KojiBuild> builds, String username) {
```

Comment:
Refactor this function by splitting into smaller functions, please.

Reply:
OK, I have added six new methods, but maybe it's still kind of big.

Reply:
OK, should be small now.